### PR TITLE
Change ListProfilesByBundleIdCommand to use fuzzy search

### PR DESCRIPTION
**Pull request by DechengMa**
Remove bundle id must be unique validation for `ListProfilesByBundleIdCommand`.
Input first part of the bundle ID will result in all the matched cases.

# 📝 Summary of Changes

Changes proposed in this pull request:

- Removed bundle id `must be unique validation` for `ListProfilesByBundleIdCommand`.
- Return all the matched profiles for input bundle Id

# ⚠️ Items of Note

Inputted bundle Id string must be the first part of the bundle Id.

# 🧐🗒 Reviewer Notes

## 💁 Example

USAGE: 
```
asc profiles list-by-bundle [--api-issuer <uuid>] [--api-key-id <keyid>] [--csv] [--json] [--table] [--yaml] [--verbose] <bundle-id> [--limit <limit>] [--download-path <path>]
```

## 🔨 How To Test

```
swift run asc profiles list-by-bundle com --download-path ../../
```

### CODE REVIEW

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@ private typealias Filter = DownloadSalesAndTrendsReports.Filter
 extension Filter.Frequency: ExpressibleByArgument, CustomStringConvertible {
     private typealias AllCases = [Filter.Frequency]
 
-    public static var allCases: AllCases {
+    private static var allCases: AllCases {
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:12]: This seems unrelated?

Comment (DechengMa) [reply Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:12]: > This seems unrelated?

This is for CLI project to be running under Xcode 12. I don't know why it's required but Xcode wouldn't compile without this

Comment (orj) [reply Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:12]: Right.  Yeah. I noticed earlier actually. Was going to ask.

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@ private typealias Filter = DownloadSalesAndTrendsReports.Filter
 extension Filter.Frequency: ExpressibleByArgument, CustomStringConvertible {
     private typealias AllCases = [Filter.Frequency]
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:10]: You could probably fix this by changing:
```suggestion
    public typealias AllCases = [Filter.Frequency]
```

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift (hunk 1 above)
Comment (orj) [reply Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:12]: If you changed the type alias to public you could leave the other stuff public.

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift [hunk 3]
```diff
@@ -25,7 +25,7 @@ extension Filter.Frequency: ExpressibleByArgument, CustomStringConvertible {
 extension Filter.ReportType: ExpressibleByArgument, CustomStringConvertible {
     private typealias AllCases = [Filter.ReportType]
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:26]:
```suggestion
    public typealias AllCases = [Filter.ReportType]
```

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift [hunk 4]
```diff
@@ -25,7 +25,7 @@ extension Filter.Frequency: ExpressibleByArgument, CustomStringConvertible {
...
 
-    public static var allCases: AllCases {
+    private static var allCases: AllCases {
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:28]:
```suggestion
    public static var allCases: AllCases {
```

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift [hunk 5]
```diff
@@ -41,7 +41,7 @@ extension Filter.ReportType: ExpressibleByArgument, CustomStringConvertible {
 extension Filter.ReportSubType: ExpressibleByArgument, CustomStringConvertible {
     private typealias AllCases = [Filter.ReportSubType]
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:42]:
```suggestion
    public typealias AllCases = [Filter.ReportSubType]
```

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift [hunk 6]
```diff
@@ -41,7 +41,7 @@ extension Filter.ReportType: ExpressibleByArgument, CustomStringConvertible {
...
 
-    public static var allCases: AllCases {
+    private static var allCases: AllCases {
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift:44]:
```suggestion
    public static var allCases: AllCases {
```

**Review by orj — COMMENTED** [review]
The `typelias` is what should be public perhaps.

File: Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift (hunk 1 above)
Comment (DechengMa) [reply Sources/AppStoreConnectCLI/Model/API/Sales.Filter+ExpressibleByArgument.swift]: just changed it to public and pushed ✅

**Comment by orj** [comment]
If you could update the documentation on the `list-by-bundle` command too to reflect the change in behaviour. That would be great.

**Comment by DechengMa** [comment]
> If you could update the documentation on the `list-by-bundle` command too to reflect the change in behaviour. That would be great.

Cool, will do, what do you think of this?
```
abstract: "Find and list provisioning profiles and download their data by a fuzzy search for bundleId.",
```

**Comment by orj** [comment]
I think the behaviour should be slightly different too.  I think it should only match the beginning of the bundle ID.  We'll have to do some local filtering for that.

**Comment by orj** [comment]
I think the abstract is ok as is.  I think we should add further discussion to the `bundleId` argument:

```
@Argument(help: "The bundle ID of an application. (eg. com.example.app)",
              discussion: "Note that all provisioning profiles for bundle identifiers beginning with this string will be fetched.")
    var bundleId: String
```

File: Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift [hunk 7]
```diff
@@ -703,19 +703,23 @@ class AppStoreConnectService {
...
-            options: .init(bundleId: bundleId)
+        let bundleIdResourceIds = try ListBundleIdsOperation(
+            options: .init(identifiers: [bundleId], names: [], platforms: [], seedIds: [], limit: nil)
         )
-            .execute(with: requestor)
-            .await()
-            .id
+        .execute(with: requestor)
+        .await()
+        .filter { ($0.attributes?.identifier?.starts(with: bundleId) ?? false) }
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift:711]: Nice 👍

**Review by orj — APPROVED** [review]
Nice work @DechengMa. Thanks.
